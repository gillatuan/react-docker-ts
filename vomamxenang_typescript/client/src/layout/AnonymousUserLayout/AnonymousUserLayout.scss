@import 'styles/variables.scss';

.anonymous-layout {
  &-content {
    width: $anonymous-width-content;
  }
}
