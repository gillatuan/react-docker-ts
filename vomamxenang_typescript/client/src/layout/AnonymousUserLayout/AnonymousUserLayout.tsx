import React from 'react';
import 'layout/AnonymousUserLayout/AnonymousUserLayout.scss';

const AnonymousUserLayout: React.FC = ({ children }) => {
  return (
    <div className="fullscreen anonymous-layout">
      <div className="fullscreen__section fullscreen__section--center">
        <div className="fullscreen__section__child anonymous-layout-content">{children}</div>
      </div>
    </div>
  );
};

export default AnonymousUserLayout;
