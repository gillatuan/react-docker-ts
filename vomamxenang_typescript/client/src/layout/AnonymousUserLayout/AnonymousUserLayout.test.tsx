import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import AnonymousUserLayout from 'layout/AnonymousUserLayout/AnonymousUserLayout';

describe('<AnonymousUserLayout />', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <AnonymousUserLayout>
        <div>Hi, this is some form for anonymous user like login, register, etc.</div>
      </AnonymousUserLayout>
    </BrowserRouter>
  );

  test('render snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
