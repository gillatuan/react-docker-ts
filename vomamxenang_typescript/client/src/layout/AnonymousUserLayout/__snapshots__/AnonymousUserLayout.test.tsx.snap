// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnonymousUserLayout /> render snapshot correctly 1`] = `
<div
  className="fullscreen anonymous-layout"
>
  <div
    className="fullscreen__section fullscreen__section--center"
  >
    <div
      className="fullscreen__section__child anonymous-layout-content"
    >
      <div>
        Hi, this is some form for anonymous user like login, register, etc.
      </div>
    </div>
  </div>
</div>
`;
